services/QuillLMS/app/models/canvas_account.rb

```diff
@@ -22,10 +22,42 @@
 #  fk_rails_...  (user_id => users.id)
 #
 class CanvasAccount < ApplicationRecord
+  class InvalidUserExternalIdFormatError < StandardError; end
+
+  VALID_USER_EXTERNAL_ID_FORMAT = /\A\d+:[a-fA-F0-9]+\z/.freeze
+
   belongs_to :canvas_instance
   belongs_to :user
 
-  def canvas_id
+  def self.build_user_external_id(canvas_instance_id, external_id)
     [canvas_instance_id, external_id].join(':')
   end
+
+  def self.custom_create_by_user_external_id!(user_external_id, user_id)
+    canvas_instance_id, external_id = unpack_user_external_id!(user_external_id)
+
+    create!(canvas_instance_id: canvas_instance_id, external_id: external_id, user_id: user_id)
+  end
+
+  def self.custom_find_by_user_external_id(user_external_id)
+    return nil unless valid_user_external_id_format?(user_external_id)
+
+    canvas_instance_id, external_id = user_external_id.split(':')
```

Comment:
Should this be a call to `unpack_user_external_id` instead of `split`?

Reply:
I'm not using `unpack_user_external_id!` since I wanted a split method that did not raise an error on exception.